### ISSUE TITLE
Hardcoded icon path in launcher for GNU/Linux systems

### ISSUE THREAD

**Issue by palob** [Komodo/KomodoEdit#308]
Hi,

currently the icon path specified in the created [.desktop launcher file](https://github.com/Komodo/KomodoEdit/blob/8e746534eed6b32d40676375fbfe0c90f1814bb4/src/install/_install.p.py#L287) is set to:
``Icon=%s/share/icons/komodo48.png``

This prevents icon theming without editing the launcher file.

I'd like to ask you to change the icon location as well as the icon path line in the launcher according to the freedesktop.org standards.

Please find more info here:
http://standards.freedesktop.org/icon-theme-spec/icon-theme-spec-latest.html
https://github.com/Foggalong/hardcode-fixer/wiki/What,-Why-&-How

According to the above resources standard icon locations are:

``/usr/share/icons/hicolor/[size]/apps/[icon name]``
``~/.local/share/icons/hicolor/[size]/apps/[icon name]``
or simply:
``/usr/share/pixmaps/[icon name]``

The corresponding standard icon line in the .desktop launcher is:
``Icon=[icon name]`` (without full path, without file extension)

for instance
``Icon=komodo``

(for an icon ``komodo.svg`` or ``komodo.png`` stored in standard location

Could you please install the icon to one of those locations and adjust the icon path in the .desktop file accordingly?

**Comment by Naatan** [comment]
Aye this has annoyed me too - will do.

**Comment by Naatan** [comment]
Reading up on this, is there no way to use this without installing an icon in an OS directory? We are not packaging Komodo for individual linux distro's atm, which means Komodo comes with all its dependencies and doesn't "infect" directories outside of its install and profile directories. It seems to me that's a problem with defining an icon that can be picked up by icon themes while still being able to fall back on the default, unless Komodo extracts its icon to the OS icon dir, which simply isnt feasible unless we package Komodo for individual distro's.

**Comment by palob** [comment]
Well, [your chosen icon path](https://github.com/Komodo/KomodoEdit/blob/8e746534eed6b32d40676375fbfe0c90f1814bb4/src/install/_install.p.py#L287) allows for both a global/local installation into ``/usr/share/...`` or ``$HOME/.local/share/...`` respectively.
It's already stored outside of Komodo's install and profile directories.

Reading the paragraph in http://standards.freedesktop.org/icon-theme-spec/icon-theme-spec-latest.html about the icon lookup procedure it occured to me ``$XDG_DATA_DIRS/icons`` is another standard directory. Hence it might very well be possible to change the [icon line](https://github.com/Komodo/KomodoEdit/blob/8e746534eed6b32d40676375fbfe0c90f1814bb4/src/install/_install.p.py#L287) to simply ``Icon=komodo48`` (or ``Icon=komodo`` if the PNG is renamed to ``komodo.png``) without even touching the path the icon is installed to. It will show up no matter what's the chosen icon theme.

However for some reason the preferred paths are ``.../share/icons/hicolor/[size]/apps/[icon name]`` or simply ``.../share/pixmaps``. At least that's how other application do this, may be not to clutter the ``icons`` directory.

The ´´hicolor´´ theme is standardised by *freedesktop.org* and AFAIK each and every icon theme coming with whatever distro inherits from it. Otherwise many application icons wouldn't show up.

TL;DR
Following the *freedesktop.org* standard actually is what ensures that icons > can be picked up by icon themes while still being able to fall back on the default

**Comment by palob** [comment]
One more argument in favour of "dehardcoding":

Currently the hardcoded icon line ``Icon=%s/share/icons/komodo48.png`` causes that only the ``48px X 48px`` will be shown.

Whenever an icon bigger than that is needed (e.g. GNOME Shell overview, hidpi screens) the icon will appear blurry.
Whenever an icon smaller than that is needed you will lose on recognisability compared to an icon designed for smaller sizes.

However you do provide icon in various sizes from as small as 16x16 up to 256x256.

By changing the icon line to ``Icon=komodo``, installing the respectively sized icons to ``.../share/icons/hicolor/[size]/apps/`` and renaming all to just ``komodo.png`` you can make sure that always the appropriately sized icon is shown.

It's specified this way in the [``index.theme``](https://cgit.freedesktop.org/xdg/default-icon-theme/tree/index.theme) of the ``hicolor`` icon theme.

The ``hicolor`` theme is present in all desktop environment adhering to [freedesktop.org](https://www.freedesktop.org/) standards. All DEs available for GNU/Linux or BSD I know of (GNOME, KDE Plasma, Xfce, MATE, LXDE, LXQt, Pantheon, Cinnamon, Unity, Budgie, Deepin, Enlightenment, Lumina, Hawaii Shell, Papyros Shell) do this.

**Comment by Naatan** [comment]
I'll try experimenting with this, but if the install directory itself cannot be used as a fallback then this is a no-go at least until we do distro specific packages.
